src/Nethermind/Nethermind.Runner/Ethereum/Steps/InitializeNetwork.cs

```diff
@@ -400,67 +397,12 @@ private async Task InitPeer()
-                if (_api.NdmDataPublisher == null) throw new StepDependencyException(nameof(_api.NdmDataPublisher));
-                if (_api.NdmConsumerChannelManager == null) throw new StepDependencyException(nameof(_api.NdmConsumerChannelManager));
-                if (_api.BloomStorage == null) throw new StepDependencyException(nameof(_api.BloomStorage));
-                if (_api.ReceiptFinder == null) throw new StepDependencyException(nameof(_api.ReceiptFinder));
-
-                if (_logger.IsInfo) _logger.Info($"Initializing NDM...");
-                _api.HttpClient = new DefaultHttpClient(new HttpClient(), _api.EthereumJsonSerializer, _api.LogManager);
-                INdmConfig ndmConfig = _api.Config<INdmConfig>();
-                if (ndmConfig.ProxyEnabled)
-                {
-                    _api.JsonRpcClientProxy = new JsonRpcClientProxy(_api.HttpClient, ndmConfig.JsonRpcUrlProxies,
-                        _api.LogManager);
-                    _api.EthJsonRpcClientProxy = new EthJsonRpcClientProxy(_api.JsonRpcClientProxy);
-                }
-
-                FilterStore filterStore = new FilterStore();
-                FilterManager filterManager = new FilterManager(filterStore, _api.MainBlockProcessor, _api.TxPool, _api.LogManager);
-                INdmCapabilityConnector capabilityConnector = await _api.NdmInitializer.InitAsync(
-                    _api.ConfigProvider,
-                    _api.DbProvider,
-                    initConfig.BaseDbPath,
-                    _api.BlockTree,
-                    _api.TxPool,
-                    _api.TxSender,
-                    _api.SpecProvider,
-                    _api.ReceiptFinder,
-                    _api.Wallet,
-                    filterStore,
-                    filterManager,
-                    _api.Timestamper,
-                    _api.EthereumEcdsa,
-                    _api.RpcModuleProvider,
-                    _api.KeyStore,
-                    _api.EthereumJsonSerializer,
-                    _api.CryptoRandom,
-                    _api.Enode,
-                    _api.NdmConsumerChannelManager,
-                    _api.NdmDataPublisher,
-                    _api.GrpcServer,
-                    _api.NodeStatsManager,
-                    _api.ProtocolsManager,
-                    protocolValidator,
-                    _api._messageSerializationService,
-                    initConfig.EnableUnsecuredDevWallet,
-                    _api.WebSocketsManager,
-                    _api.LogManager,
-                    _api.MainBlockProcessor,
-                    _api.JsonRpcClientProxy,
-                    _api.EthJsonRpcClientProxy,
-                    _api.HttpClient,
-                    _api.MonitoringService,
-                    _api.BloomStorage);
-
-                capabilityConnector.Init();
-                if (_logger.IsInfo) _logger.Info($"NDM initialized.");
+                await plugin.InitNetworkProtocol();
```

Comment:
probably we need dependency on InitializePlugins step

Reply:
fixed